src/lib/dnssd/ActiveResolveAttempts.h

```diff
@@ -181,6 +186,28 @@ class ActiveResolveAttempts
         bool IsIpResolve() const { return resolveData.Is<IpResolve>(); }
         void Clear() { resolveData = DataType(); }
 
+        void WillCoalesceWith(const ScheduledAttempt & existing)
+        {
+            if (IsResolve() && existing.Matches(*this))
```

Comment:
This method needs some documentation about what its purpose and use is.
I am also wondering of the `Matches` is not met - what does that mean and should we silently ignore the overwrite? What sets consumerCount to 0/1 to start with?

Also to decrease indent and help documentation we could split ifs:

```cpp
if (!IsResolve()) {
    return; // consumer count is only tracked for resolve requests
}

if (!existing.Matches(*this)) {
   // Adding another consumer to the same query, 
   // TODO: what if this does NOT match? is this overwriting? should we log an error?
   return; 
}

resolveData.Get<Resolve>().consumerCount++;
```

Reply:
> This method needs some documentation about what its purpose and use is.

Will do.

> I am also wondering of the Matches is not met - what does that mean and should we silently ignore the overwrite? 

If it's not met, the new attempt is for a different resolve and it has a different set of consumers.  We already log an error for this case in `ActiveResolveAttempts::MarkPending`:
```
        // TODO: node was evicted here, if/when resolution failures are
        // supported this could be a place for error callbacks
        //
        // Note however that this is NOT an actual 'timeout' it is showing
        // a burst of lookups for which we cannot maintain state. A reply may
        // still be received for this peer id (query was already sent on the
        // network)
        ChipLogError(Discovery, "Re-using pending resolve entry before reply was received.");
```

I will add some better documentation.

> What sets consumerCount to 0/1 to start with?

The `Resolve` struct constructor, since it's declared as:
```
            uint32_t consumerCount = 0;
```

> Also to decrease indent and help documentation we could split ifs:

Done.